docs_src/design/adr/application/0025-Record-and-Replay.md

```diff
@@ -0,0 +1,128 @@
+# Record and Replay ADR
+
+## Submitters
+- Lenny Goodell (Intel)
+
+## Change Log
+- [Pending](https://github.com/edgexfoundry/edgex-docs/pull/863) (2022-09-20)
+
+## Referenced Use Case(s)
+- [Record and Replay UCR](https://docs.edgexfoundry.org/2.3/design/ucr/Record-and-Replay/)
+
+## Context
+This design involves creating a new Application Service that is responsible for the requirements in the above referenced UCR. This new Application Service will go above and beyond that of a normal Application Service,  thus it is architecturally significant.
+
+## Proposed Design
+A new Application Service will be created with a RESTful API to handle the Record, Replay, Export and Import capabilities. The service will not create or start a Functions Pipeline on start-up as normally done in Application Services. It will wait until the Record request has been received. Once the recording is complete the Functions Pipeline will be stopped. There will be a very rudimentary HTML page to serve as UI to access this service RESTful API.
```

Comment:
Concern with bringing in packages to provide any type of display and not sure it is needed if you have the APIs (and postman).  But I defer to the community on this one.

Reply:
Thinking about the "rudimentary HTML page" that our product has for a similar recorder, we do have CGI scripts (with user role checking) behind the simple push buttons, so I suppose it can't be too rudimentary. I'd support dropping this, too, then.

Reply:
I am good with this!!!